ai_engine.py

```diff
@@ -141,56 +141,56 @@ def evaluate_board(self, game_state, player):
+            if piece.is_player("white"):
+                if piece.get_name() == "k":
                     return -1000
-                elif piece.get_name() is "q":
+                elif piece.get_name() == "q":
                     return -100
-                elif piece.get_name() is "r":
+                elif piece.get_name() == "r":
                     return -50
-                elif piece.get_name() is "b":
+                elif piece.get_name() == "b":
                     return -30
-                elif piece.get_name() is "n":
+                elif piece.get_name() == "n":
                     return -30
-                elif piece.get_name() is "p":
+                elif piece.get_name() == "p":
                     return -10
             else:
-                if piece.get_name() is "k":
+                if piece.get_name() == "k":
                     return 1000
-                elif piece.get_name() is "q":
+                elif piece.get_name() == "q":
                     return 100
-                elif piece.get_name() is "r":
+                elif piece.get_name() == "r":
                     return 50
-                elif piece.get_name() is "b":
+                elif piece.get_name() == "b":
                     return 30
-                elif piece.get_name() is "n":
+                elif piece.get_name() == "n":
                     return 30
-                elif piece.get_name() is "p":
+                elif piece.get_name() == "p":
                     return 10
         else:
             if piece.is_player("white"):
-                if piece.get_name() is "k":
+                if piece.get_name() == "k":
                     return 1000
-                elif piece.get_name() is "q":
+                elif piece.get_name() == "q":
                     return 100
-                elif piece.get_name() is "r":
+                elif piece.get_name() == "r":
                     return 50
-                elif piece.get_name() is "b":
+                elif piece.get_name() == "b":
                     return 30
-                elif piece.get_name() is "n":
+                elif piece.get_name() == "n":
                     return 30
-                elif piece.get_name() is "p":
+                elif piece.get_name() == "p":
                     return 10
             else:
-                if piece.get_name() is "k":
+                if piece.get_name() == "k":
                     return -1000
-                elif piece.get_name() is "q":
+                elif piece.get_name() == "q":
                     return -100
-                elif piece.get_name() is "r":
+                elif piece.get_name() == "r":
                     return -50
-                elif piece.get_name() is "b":
+                elif piece.get_name() == "b":
                     return -30
-                elif piece.get_name() is "n":
+                elif piece.get_name() == "n":
                     return -30
```

Comment:
This is not the problem of the code, the usage of 'is' will give a **syntex warning**, try again

Reply:
I didn't change it as the problem of the code, just Pycharm didn't like it - throwed an error (I think that by the new versions he doesn't exepts 'is' just '=='). besides that, I have fuxed the 3 problems of the code